partitionccl: unskip TestRepartitioning

aadityasondhi opened the pull request:
Most of the changes were cherry picked from
https://github.com/cockroachdb/cockroach/pull/69424.

Closes #49112.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105823)
<!-- Reviewable:end -->

aadityasondhi committed:
partitionccl: unskip TestRepartitioning

Most of the changes were cherry picked from
https://github.com/cockroachdb/cockroach/pull/69424.

Closes #49112.

Release note: None
